rust/src/lib/nm/query_apply/dispatch.rs

```diff
@@ -115,6 +115,7 @@ fn write_execute_file(file_path: &str, content: &str) -> std::io::Result<()> {
         .mode(0o744)
         .open(file_path)?;
     fd.write_all(content.as_bytes())?;
+    fd.sync_all()?;
```

Comment:
Why is it needed to ensure the file is written to disk? That is normally not necessary for inter-process synchronization: you can just write to the file, close it (or flush it) and then other processes can see the changes. Waiting for sync_all() adds an unnecessary and potentially long delay.